spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -470,6 +470,7 @@ private void stopInvokerAndCommitManualAcks() {
 			}
 			// handle the last manual acks, after the listeners have closed
 			handleManualAcks();
+			processCommits();
```

Comment:
Since we are adding this one here we don't need the next `commitIfNecessary()` with its `if`.
Minor, of course, because after `processCommits()` we will never reach that  `commitIfNecessary()`: `this.offsets` is empty already.